app/RecordStatus.php

```diff
@@ -78,6 +102,150 @@ public static function activate(string $moduleName, string $fieldName): bool
+	 */
+	public static function getTimeCountingArrayValueFromString(string $timeCountingStr): array
+	{
+		if ($timeCountingStr === '' || $timeCountingStr === ',,') {
+			return [];
+		}
+		if (strpos($timeCountingStr, ',') === -1) {
+			throw new \App\Exceptions\IllegalValue('ERR_NOT_ALLOWED_VALUE||' . $timeCountingStr, 406);
+		}
+		$values = explode(',', trim($timeCountingStr, ','));
+		if (!$values) {
+			return [];
+		}
+		$result = [];
+		foreach ($values as $value) {
+			if (!is_numeric($value)) {
+				throw new \App\Exceptions\IllegalValue('ERR_NOT_ALLOWED_VALUE||' . $value, 406);
+			}
+			$result[] = (int) $value;
+		}
+		return $result;
+	}
+
+	/**
+	 * Get time counting value from array.
+	 *
+	 * @param int[] $timeCountingArr
+	 *
+	 * @throws \App\Exceptions\IllegalValue
+	 *
+	 * @return string
+	 */
+	public static function getTimeCountingStringValueFromArray(array $timeCountingArr): string
+	{
+		foreach ($timeCountingArr as $time) {
+			if (!is_int($time)) {
+				throw new \App\Exceptions\IllegalValue('ERR_NOT_ALLOWED_VALUE||' . $time, 406);
+			}
+		}
+		return ',' . implode(',', $timeCountingArr) . ',';
+	}
+
+	/**
+	 * Get closing states for all fields in module.
+	 *
+	 * @param string $moduleName
+	 * @param bool   $byName
+	 *
+	 * @return string[]
+	 */
+	public static function getCloseStates(string $moduleName, bool $byName = true)
+	{
+		$tabId = \App\Module::getModuleId($moduleName);
+		$cacheName = 'RecordStatus::getCloseStates' . ($byName ? 'ByName' : '');
+		if (\App\Cache::staticHas($cacheName, $tabId)) {
+			return \App\Cache::staticGet($cacheName, $tabId);
+		}
+		$field = $byName ? ['vtiger_field.fieldname', 'value'] : ['valueid', 'value'];
+		$values = (new \App\Db\Query())->select($field)
+			->from('u_#__picklist_close_state')
+			->innerJoin('vtiger_field', 'u_#__picklist_close_state.fieldid = vtiger_field.fieldid')
+			->where(['tabid' => $tabId, 'presence' => [0, 2]])
+			->createCommand()->queryAllByGroup($byName ? 2 : 0);
+		\App\Cache::staticSave($cacheName, $tabId, $values);
+		return $values;
+	}
+
+	/**
+	 * Get all record states.
+	 *
+	 * @return string[] [id=>label]
+	 */
+	public static function getLabels(): array
+	{
+		return [self::RECORD_STATE_NO_CONCERN => 'LBL_RECORD_STATE_NO_CONCERN', self::RECORD_STATE_OPEN => 'LBL_RECORD_STATE_OPEN', self::RECORD_STATE_CLOSED => 'LBL_RECORD_STATE_CLOSED'];
+	}
+
+	/**
+	 * Get process status field names.
+	 *
+	 * @param string $moduleName optional if we need only one field name for specified module
+	 *
+	 * @return string|string[]
+	 */
+	public static function getFieldName(string $moduleName = '')
+	{
+		if (\App\Cache::has('RecordStatus::getFieldName', $moduleName)) {
+			return \App\Cache::get('RecordStatus::getFieldName', $moduleName);
+		}
+		$query = (new \App\Db\Query())
+			->select(['vtiger_field.fieldname', 'vtiger_field.tabid'])
+			->from('vtiger_field')
+			->where(['fieldparams' => '{"isProcessStatusField":true}', 'presence' => [0, 2]]);
```

Comment:
'{"isProcessStatusField":true}' It works, but only now

Reply:
yap, before was `['LIKE',$column,'"isProcessStatusField":true']` so even with other params it will find it